synapse/lib/stormlib/model.py

```diff
@@ -1268,3 +1319,326 @@ async def _riskHasVulnToVulnerable(self, n, nodata=False):
+                for $ref in $n.refs {
+                    ($form, $prop, $type, $isarray) = $ref.refinfo
+                    $lib.print(`    - {$form}:{$prop} (iden: {$ref.iden}) (view: {$ref.view})` )
+                }
+            }
+
+            if $n.edges {
+                $lib.print(`  edges:`)
+                for $edge in $n.edges {
+                    if ($edge.direction = "n2") {
+                        $lib.print(`      <({$edge.verb})- {$edge.node} (view: {$edge.view})`)
+                    } else {
+                        $lib.print(`      -({$edge.verb})> {$edge.node} (view: {$edge.view})`)
+                    }
+                }
+            }
+        }
+
+        function _checkValidLayer(viewiden, layriden) {
+            try {
+                $view = $lib.view.get($viewiden)
+            } catch NoSuchView as exc {
+                $lib.warn(`View {$viewiden} does not exist.`)
+                return($lib.false)
+            }
+
+            $wlyr = $view.layers.0
+
+            if ($wlyr.iden != $layriden) {
+                $lib.warn(`Layer mismatch for node: expected {$layriden}, got {$wlyr.iden}.`)
+                return($lib.false)
+            }
+
+            if (not $lib.user.allowed("node", gateiden=$wlyr.iden)) {
+                $lib.warn(`User {$lib.user.name()} does not have write access to view {$viewiden}.`)
+                return($lib.false)
+            }
+
+            return($lib.true)
+        }
+
+        function _repairNode(iden, newval) {
+            $oldnode = $lib.model.migration.s.model_0_2_31.queueData.$iden
+            if (not $oldnode) {
+                $lib.warn(`Queue node not found with iden {$iden}.`)
+                return($lib.false)
+            }
+
+            if (not $_checkValidLayer($oldnode.view, $oldnode.layer)) {
+                $lib.warn("View/layer does not exist to recreate node.")
+                return($lib.false)
+            }
+
+            ($ok, $norm) = $lib.trycast($oldnode.form, $newval)
+            if (not $ok) {
+                $lib.warn(`Invalid value provided for {$oldnode.form}: {$newval}.`)
+                return($lib.false)
+            }
+
+            $lib.print(`Repairing {$iden} from {$oldnode.valu} -> {$newval}.`)
+
+            // First thing, create the node in the right view
+            view.exec $oldnode.view {
+                [ *$oldnode.form = $newval ]
+            }
+
+            for $edit in $oldnode.edits {
+                if (not $_checkValidLayer($edit.view, $edit.layer)) {
+                    $lib.warn(`Could not apply edits in view {$edit.view}, skipping.`)
+                    continue
+                }
+
+                view.exec $edit.view {
+                    *$oldnode.form=$newval
+
+                    { for ($propname, $propvalu) in $edit.props {
+                        $valu = $propvalu.0
+
+                        if ($propname = ".seen") {
+                            [ .seen = $valu ]
```

Comment:
Don't actually need to special case `.seen` here `:.seen` is interpreted correctly as a univ prop